File: langgraph/data/equiv/128.py
def calculate_signed_sum(numbers):
    if not numbers:
        return None
    
    negative_count = sum(1 for num in numbers if num < 0)
    contains_zero = any(num == 0 for num in numbers)
    
    sign = 0 if contains_zero else (-1) ** negative_count
    total_sum = sum(abs(num) for num in numbers)
    
    return sign * total_sum

# method name: prod_signs -> calculate_signed_sum
# parameter name: arr -> numbers
# variable name: prod -> sign, used a more descriptive name
# replaced filter and lambda with a generator expression for negative count
# replaced list comprehension with sum and generator expression for total_sum
# changed the logic to use any() for checking zero presence
# kept the overall logic but restructured the code for clarity

File: langgraph/result/qwen/32b-eq2js/128.js
function calculateSignedSum(numbers) {
    if (!numbers.length) {
        return null;
    }
    
    const negativeCount = numbers.reduce((count, num) => count + (num < 0 ? 1 : 0), 0);
    const containsZero = numbers.some(num => num === 0);
    
    const sign = containsZero ? 0 : Math.pow(-1, negativeCount);
    const totalSum = numbers.reduce((sum, num) => sum + Math.abs(num), 0);
    
    return sign * totalSum;
}